src/mlte/artifact/model.py

```diff
@@ -0,0 +1,37 @@
+"""
+mlte/artifact/model.py
+
+Model implementation for MLTE artifacts.
+"""
+
+from enum import Enum
+
+from mlte.model import BaseModel
+from mlte.negotiation.model import NegotiationCardModel
+
+
+class ArtifactType(str, Enum):
+    """Enumerates all supported artifact types."""
+
+    NEGOTIATION_CARD = "negotiation_card"
+    """The negotiation card artifact type."""
+
+
+class ArtifactHeaderModel(BaseModel):
+    """The NegotiationCardHeaderModel contains the common metadata for all artifacts."""
+
+    identifier: str
+    """The unique identifier for the artifact."""
+
+    type: ArtifactType
+    """The type identfier for the artifact."""
+
+
+class ArtifactModel(BaseModel):
+    """The base model for all MLTE artifacts."""
+
+    header: ArtifactHeaderModel
+    """The artifact header."""
+
+    body: NegotiationCardModel
```

Comment:
This is confusing. My plan here is to make the body a `Union` over all of the various artifact types, so this will be:

```python
header: ArtifactHeaderModel

body: Union[SpecModel, NegotiationCardModel, ...]
```

The `Union` annotation just won't allow a `Union` with a single type, and the `NegotiationCardModel` is the only one implemented at present.

Reply:
Roger, updated this with a comment.